types/twine-sugarcube/save.d.ts

```diff
@@ -33,6 +33,44 @@ export interface SaveObject {
     version?: any;
 }
 
+export interface SaveDetails {
+    /**
+     * A string representing how the save operation came about — i.e., what caused it.
+     */
+    type: 'autosave' | 'disk' | 'serialize' | 'slot';
+}
+
+type SaveHandler = (save: SaveObject, details: SaveDetails) => void;
+type LoadHandler = (save: SaveObject) => void;
+
+interface SaveEventAPI<HandlerType> {
+    /**
+     * Add new handler
+     * @param handler
+     */
+    add(handler: HandlerType): void;
+    /**
+     * Deletes all currently registered handlers.
+     */
+    clear(): void;
+    /**
+     * Deletes the specified handler.
+     * @param handler The handler function to be deleted.
+     * @example
+     * // Given:
+     * // 	let myOnLoadHandler = function (save) {
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/twine-sugarcube/save.d.ts;
  Line: 61;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.